caps/capabilities.go

```diff
@@ -0,0 +1,104 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"errors"
+	"regexp"
+	"sort"
+)
+
+// Type defines the "kind" of a capability.
+// Each type defines the behavior allowed and expected from providers
+// and consumers of that capability, and also which information should
+// be exchanged by these parties.
+type Type string
+
+// Capability holds information about a capability that a snap may request
+// from a snappy system to do its job while running on it.
+type Capability struct {
+	// Name is a key that identifies the capability. It must be unique within its
+	// context, which may be either a snap or a snappy runtime.
+	Name string
+	// Label provides an optional title for the capability to help a human tell
+	// which physical device this capability is referring to. It might say
+	// "Front USB", or "Green Serial Port", for example.
+	Label string
+	// CapabilityType describes a group of capabilities sharing some common
```

Comment:
This is still out of date in comparison to the review done in the original branch.

By the way, why was this review taken away from the original pull request?  Usually reviews are applied on the branch where comments have been made, before merging.